edpop_explorer/readers/stcn.py

```diff
@@ -1,104 +1,241 @@
+    def _get_format(cls, rawrecord: dict) -> Optional[Field]:
+        format_ = safeget(rawrecord, ("data", "format", "format"))
+        if format_ is None:
+            return None
+        return Field(format_)
+
+    @classmethod
+    def _get_collation_formula(cls, rawrecord: dict) -> Optional[Field]:
+        collations = safeget(rawrecord, ("data", "extent", "collation"))
+        if not collations:
+            return None
+        # Multiple collation formulas are possible, but this seems to be rare.
+        collation_string = ' ; '.join([x.get("value") for x in collations if "value" in x])
+        return Field(collation_string)
+
+    @classmethod
+    def _get_fingerprint(cls, rawrecord: dict) -> Optional[Field]:
+        fingerprints = safeget(rawrecord, ("data", "fingerprint"))
+        if not fingerprints:
+            return None
+        # Multiple fingerprints are possible, but this seems to be rare
+        fingerprint_string = ' ; '.join([x.get("fingerprint") for x in fingerprints if "fingerprint" in x])
+        return Field(fingerprint_string)
+
+    @classmethod
+    def _get_genres(cls, rawrecord: dict) -> List[Field]:
+        subjecttopics = safeget(rawrecord, ("data", "subjectTopic"))
+        if subjecttopics is None:
+            return []
+        fields = [Field(x["preferred"]) for x in subjecttopics if "preferred" in x]
+        return fields
+
+    @classmethod
+    def _get_holdings(cls, rawrecord: dict) -> List[Field]:
+        holdings = safeget(rawrecord, ("data", "holdings"))
+        if holdings is None:
+            return []
+        fields = []
+        for holding in holdings:
+            institution = safeget(holding, ("data", "institutionName"))
+            shelfmark = safeget(holding, ("data", "shelfmark"))
+            summary = f"{institution} - {shelfmark}"
+            fields.append(Field(summary))
+        return fields
+
+    @classmethod
+    def _convert_record(cls, rawrecord: dict) -> BibliographicalRecord:
+        record = BibliographicalRecord(from_reader=cls)
+        record.data = rawrecord
+        record.identifier = rawrecord.get('id', None)
+        if record.identifier:
+            record.link = cls.LINK_BASE_URL + record.identifier
+        record.title = cls._get_title(rawrecord)
+        record.contributors = cls._get_contributors(rawrecord)
+        record.publisher_or_printer = cls._get_publisher_or_printer(rawrecord)
+        record.place_of_publication = cls._get_place_of_publication(rawrecord)
+        record.dating = cls._get_dating(rawrecord)
+        record.languages = cls._get_languages(rawrecord)
+        record.extent = cls._get_extent(rawrecord)
+        record.bibliographical_format = cls._get_format(rawrecord)
+        record.collation_formula = cls._get_collation_formula(rawrecord)
+        record.fingerprint = cls._get_fingerprint(rawrecord)
+        record.genres = cls._get_genres(rawrecord)
+        record.holdings = cls._get_holdings(rawrecord)
```

Comment:
It's a bit unconventional to first create an instance of a class and then assign all of its attributes afterwards. More conventional would be to pass all of those attributes to the `__init__` method of the class.

Reply:
I agree. I would however incline not to do this now because this concerns all cases where a `Record` or its subclasses are filled.